packages/ccd-js-gen/src/lib.ts

```diff
@@ -205,68 +264,68 @@ async function generateCode(
         )}Contract`;
         const contractClientId = 'contractClient';
 
-        generateContractBaseCode(
-            contractSourceFile,
-            contract.contractName,
-            contractClientId,
-            contractClientType,
-            moduleRef,
-            tsNocheck,
-            contractSchema
+        contractSourceFile.addStatements(
+            generateContractBaseStatements(
+                contract.contractName,
+                contractClientId,
+                contractClientType,
+                moduleRef,
+                tsNocheck,
+                contractSchema
+            )
+        );
+        notifier.done(
+            `Generate base statements for '${contract.contractName}'.`
         );
 
         for (const entrypointName of contract.entrypointNames) {
-            const startTime = Date.now();
             const entrypointSchema =
                 contractSchema?.receive.get(entrypointName);
-            generateContractEntrypointCode(
-                contractSourceFile,
-                contract.contractName,
-                contractClientId,
-                contractClientType,
-                entrypointName,
-                entrypointSchema
+            contractSourceFile.addStatements(
+                generateContractEntrypointStatements(
+                    contract.contractName,
+                    contractClientId,
+                    contractClientType,
+                    entrypointName,
+                    entrypointSchema
+                )
+            );
+            notifier.done(
+                `Generate statements for '${contract.contractName}.${entrypointName}'.`
             );
-            const spentTime = Date.now() - startTime;
-            doneItems++;
-            notifyProgress?.({
-                type: 'Progress',
-                totalItems,
-                doneItems,
-                spentTime,
-            });
         }
     }
 }
 
 /**
  * Generate code for a smart contract module client.
- * @param moduleSourceFile The sourcefile of the module.
  * @param moduleRef The module reference.
  * @param moduleClientId The identifier to use for the module client.
  * @param moduleClientType The identifier to use for the type of the module client.
+ * @param tsNocheck When `true` generate `// @ts-nocheck` annotations in each file.
  * @param internalModuleClientId The identifier to use for referencing the internal module client.
```

Comment:
The order of the statements is swapped around.

```suggestion
 * @param internalModuleClientId The identifier to use for referencing the internal module client.
 * @param tsNocheck When `true` generate `// @ts-nocheck` annotations in each file.
```